### PR TITLE
Remove deprecated checkCanSetUser and principal rules

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: presto-plugin-toolkit/src/test/java/io/prestosql/plugin/base/security/TestFileBasedSystemAccessControl.java [hunk 1]
```diff
@@ -326,48 +316,6 @@ public void testTableRulesForCheckCanRenameTable()
         assertAccessDenied(() -> accessControl.checkCanRenameTable(ALICE, new CatalogSchemaTableName("some-catalog", "aliceschema", "alicetable"), new CatalogSchemaTableName("some-catalog", "bobschema", "newalicetable")), RENAME_TABLE_ACCESS_DENIED_MESSAGE);
     }
 
-    @Test
-    public void testCanSetUserOperations()
```
Comment (kokosing) [inline presto-plugin-toolkit/src/test/java/io/prestosql/plugin/base/security/TestFileBasedSystemAccessControl.java:330]: Don't we need to port this to `testCanImpersonateUser`? I don't see this test here. Maybe it is not needed as it is already tested elsewhere.

Comment (electrum) [reply presto-plugin-toolkit/src/test/java/io/prestosql/plugin/base/security/TestFileBasedSystemAccessControl.java:330]: `testCanImpersonateUserOperations` seems to be complete

**Comment by kokosing** [comment]
Do you think it was deprecated for enough time? Maybe we could merge this now?

**Comment by colebow** [comment]
:wave: @electrum - this PR is inactive and doesn't seem to be under development. If you'd like to continue work on this at any point in the future, feel free to re-open.

**Comment by qle1095** [comment]
curious why this PR didn't get merged..

**Pull request closed without merging**